aspnetcore/fundamentals/routing.md

```diff
@@ -731,3 +731,7 @@ routes.MapRoute("blog_route", "blog/{*slug}",
 ```
 
 Link generation only generates a link for this route when the matching values for `controller` and `action` are provided.
+
+## Complex segments
+
+Complex segments (for example [Route("/x{token}y")]) are processed by matching up literals from right to left in a non-greedy way. For detailed instructions, see the [source code](https://github.com/aspnet/Routing/blob/9cea167cfac36cf034dbb780e3f783114ef94780/src/Microsoft.AspNetCore.Routing/Patterns/RoutePatternMatcher.cs#L289).
```

Comment:
The URL should probably be https://github.com/aspnet/AspNetCore/blob/release/2.2/src/Http/Routing/src/Patterns/RoutePatternMatcher.cs#L293 as the aspnet/Routing repo has been archived.